qa/integration-tests/src/test/java/io/camunda/zeebe/it/gateway/GatewayAuthenticationIdentityIT.java

```diff
@@ -0,0 +1,260 @@
+      final var deploymentFuture =
+          client
+              .newDeployResourceCommand()
+              .addProcessModel(PROCESS_MODEL, "model.bpmn")
+              .send()
+              .toCompletableFuture();
+
+      // then
+      assertThat(deploymentFuture)
+          .failsWithin(Duration.ofSeconds(1))
+          .withThrowableOfType(ExecutionException.class)
+          .withCauseInstanceOf(StatusRuntimeException.class)
+          .extracting(
+              Throwable::getCause, as(InstanceOfAssertFactories.type(StatusRuntimeException.class)))
+          .satisfies(
+              statusRuntimeException -> {
+                assertThat(statusRuntimeException.getStatus())
+                    .hasFieldOrPropertyWithValue("code", Status.UNAUTHENTICATED.getCode())
+                    .hasFieldOrPropertyWithValue(
+                        "description", "Failed to parse bearer token, see cause for details");
+              });
+    }
+  }
+
+  @Test
+  void deployModelSucceedsWithValidAuthToken() {
+    // given
+    awaitCamundaRealmAvailabilityOnKeycloak();
+
+    try (final var client =
+        cluster
+            .newClientBuilder()
+            .defaultRequestTimeout(Duration.ofMinutes(1))
+            .credentialsProvider(
+                new OAuthCredentialsProviderBuilder()
+                    .clientId(ZEEBE_CLIENT_ID)
+                    .clientSecret(ZEEBE_CLIENT_SECRET)
+                    .audience(ZEEBE_CLIENT_AUDIENCE)
+                    .authorizationServerUrl(
+                        getKeycloakRealmAddress() + "/protocol/openid-connect/token")
+                    .build())
+            .build()) {
+      // when
+      final var deploymentFuture =
+          client
+              .newDeployResourceCommand()
+              .addProcessModel(PROCESS_MODEL, "model.bpmn")
+              .send()
+              .toCompletableFuture();
+
+      // then
+      assertThat(deploymentFuture).succeedsWithin(Duration.ofSeconds(1));
+    }
+  }
+
+  /**
+   * Awaits the presence of the Camunda realm on the keycloak container. Once Keycloak and Identity
+   * booted up, Identity will eventually configure the Camunda Realm on Keycloak.
+   */
+  private static void awaitCamundaRealmAvailabilityOnKeycloak() {
+    final var httpClient = HttpClient.newHttpClient();
+    final HttpRequest request =
+        HttpRequest.newBuilder().uri(URI.create(getKeycloakRealmAddress())).build();
+    Awaitility.await()
+        .atMost(Duration.ofSeconds(60))
+        .pollInterval(Duration.ofSeconds(5))
+        .ignoreExceptions()
+        .untilAsserted(
+            () -> {
+              final HttpResponse<String> response =
+                  httpClient.send(request, BodyHandlers.ofString());
+              assertThat(response.statusCode()).isEqualTo(200);
+            });
+  }
+
+  private static String getKeycloakRealmAddress() {
+    return "http://localhost:" + KEYCLOAK.getFirstMappedPort() + KEYCLOAK_PATH_CAMUNDA_REALM;
```

Comment:
:wrench: I think this would work

```suggestion
    return KEYCLOAK.getAuthServerUrl() + KEYCLOAK_PATH_CAMUNDA_REALM;
```

Reply:
invalid given the application of the suggested GenericContainer usage